portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,39 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public final class DataServlet extends HttpServlet {
+
+  private List<String> datas;
```

Comment:
For all variables other than local variables, prefer to declare them as final so that readers (and future debuggers) won't have to look out for where this is mutated. This is a case of preferring immutability in general: as if you mark datas as final, the List interface includes methods for mutation. If you have time, you can try to incorporate third party libraries into your project and use ImmutableList (https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableList.html) here, as that will communicate to readers that not only does datas always refer to the same list, but that list doesn't change